app/src/Components/AddScore.js

```diff
@@ -0,0 +1,83 @@
+import * as React from "react";
+
+const AddScore = ({
+  winner,
+  players,
+  addPlayer,
+  updateWin,
+  updateLoss,
+  updateTie,
+}) => {
+  const [disabledButton, setDisabledButton] = React.useState(false);
+
+  const checkWinner = async (name) => {
+    const playerNames = players.map((player) => player.name);
+    if (playerNames.includes(name)) updateWin(name);
+    else {
+      await addPlayer({ name });
+      updateWin(name);
+    }
+  };
+
+  const checkLoser = async (name) => {
+    const playerNames = players.map((player) => player.name);
+    if (playerNames.includes(name)) updateLoss(name);
+    else {
+      await addPlayer({ name });
+      updateLoss(name);
+    }
+  };
+
+  const checkTie = async (name) => {
+    const playerNames = players.map((player) => player.name);
+    if (playerNames.includes(name)) updateTie(name);
+    else {
+      await addPlayer({ name });
+      updateTie(name);
+    }
+  };
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+    const form = event.currentTarget;
+    const {
+      nameX: { value: nameX },
+      nameO: { value: nameO },
+    } = form.elements;
+
+    winner === "X"
```

Comment:
I could be misreading it, but this statement seems pretty confusing.... oh now I get it! 

Ok yeah for things with more than one case, I would probably just use normal if statements, e.g.

if (winner === "X") {
  // winner is x stuff
} else if (winner === "0") {
  // winner is o stiff
} else { // a tie
  // tie stuff
}
```